src/intentManager.js

```diff
@@ -51,167 +54,90 @@ define([
             });
         }
     };
-    ODKXIntentTag.prototype.writeXML = function (xmlWriter, currentNodeID) {
+
+    function writeXML(xmlWriter, properties) {
         xmlWriter.writeStartElement('odkx:intent');
-        xmlWriter.writeAttributeString("xmlns:odkx", this.getAttr('xmlns'));
-        xmlWriter.writeAttributeString("id", currentNodeID || this.getAttr('initialNodeID'));
-        xmlWriter.writeAttributeString("class", this.getAttr('path'));
-        _.each(this.getAttr('unknownAttributes'), function (value, name) {
+        xmlWriter.writeAttributeString("xmlns:odkx", properties.intentXmlns);
+        xmlWriter.writeAttributeString("id", properties.nodeID);
+        xmlWriter.writeAttributeString("class", properties.androidIntentAppId);
+        _.each(properties.unknownAttrs, function (value, name) {
             xmlWriter.writeAttributeString(name, value);
         });
-        writeInnerTagXML(xmlWriter, 'extra', this.getAttr('extra'));
-        writeInnerTagXML(xmlWriter, 'response', this.getAttr('response'));
+        writeInnerTagXML(xmlWriter, 'extra', properties.androidIntentExtra);
+        writeInnerTagXML(xmlWriter, 'response', properties.androidIntentResponse);
         xmlWriter.writeEndElement('odkx:intent');
-    };
-
-    var intentManager = function () {
-        var that = {};
-        that.unmappedIntentTags = {};
-
-        that.parseIntentTagsFromHead = function (tags) {
-            _.each(tags, function (tagXML) {
-                var $tag, tagId, newTag, xmlns;
-                $tag = $(tagXML);
-
-                tagId = $tag.attr('id');
-                newTag = makeODKXIntentTag(null, tagId, $tag.attr('class'));
-
-                xmlns = $tag.attr('xmlns:odkx');
-                newTag.setAttr('xmlns', xmlns || newTag.getAttr('xmlns'));
-                newTag.setAttr('extra', parseInnerTags($tag, 'extra'));
-                newTag.setAttr('response', parseInnerTags($tag, 'response'));
-                var unknowns = newTag.getAttr('unknownAttributes');
-                _.each(tagXML.attributes, function (attr) {
-                    if (attr.nodeName !== 'id' &&
-                        attr.nodeName !== 'class' &&
-                        attr.nodeName !== 'xmlns:odkx')
-                    {
-                        unknowns[attr.nodeName] = attr.nodeValue;
-                    }
-                });
-                that.unmappedIntentTags[tagId] = newTag;
-            });
-        };
+    }
 
-        that.getParsedIntentTagWithID = function (nodeID) {
-            var intentTag = null;
-            _.each(that.unmappedIntentTags, function (tag) {
-                if (tag.getAttr('initialNodeID') === nodeID) {
-                    intentTag = tag;
-                }
-            });
-            return intentTag;
-        };
+    function parseIntentTags(tags) {
+        var intentTags = {};
```

Comment:
This does cause a slight change in functionality, but I think it was a bug before.

Before if there was an unknown intent it would go into the dictionary during parsing. If you ever used 'Edit Source XML' you could go there, delete the intent, but the intent would still be in the global state, and wouldn't get deleted.

I couldn't think of another instance where loadXML would get called twice, so I felt it was safe to make the change